Cubical/Algebra/Matrix.agda

```diff
@@ -0,0 +1,65 @@
+{-# OPTIONS --cubical --safe #-}
+module Cubical.Algebra.Matrix where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Univalence
+
+import Cubical.Data.Empty as ⊥
+open import Cubical.Data.Nat hiding (_+_)
+open import Cubical.Data.Vec
+open import Cubical.Data.FinData
+open import Cubical.Relation.Nullary
+
+open import Cubical.Structures.CommRing
+
+private
+  variable
+    ℓ : Level
+
+
+-- Equivalence between Vec matrix and Fin function matrix
+
+FinMatrix : (A : Type ℓ) (m n : ℕ) → Type ℓ
+FinMatrix A m n = FinVec (FinVec A n) m
+
+VecMatrix : (A : Type ℓ) (m n : ℕ) → Type ℓ
+VecMatrix A m n = Vec (Vec A n) m
+
+FinMatrix≡VecMatrix : (A : Type ℓ) (m n : ℕ) → FinMatrix A m n ≡ VecMatrix A m n
+FinMatrix≡VecMatrix A m n i = FinVec≡Vec (FinVec≡Vec A n i) m i
+
+
+-- Experiment using addition. Transport commutativity from one
+-- representation to the the other.
+module _ (R : CommRing {ℓ}) where
+
+  open commring-operation-syntax
+
+  -- TODO: how to just use +? Seems to work for AbGroup...
```

Comment:
@borsiemir : Is it somehow possible to just write + for addition of commutative rings? It worked for AbGroup, but for CommRing I don't understand how to get it to work...

Reply:
In this case I would write:

`open commring-\cdotsyntax R`

Since we are not defining R, it is just a parameter, I wouldn't add it to the instance resoluction algorithm. But if you want to do it in that way you could use:

`open commring-\cdotsyntax {{...}}

instance
  _ : CommRing
  _ = R`

and it will also work. However I want to open an issue about what are our exact objectives with notations and improve them, but I'm afraid that this month I'll be short on time.

Reply:
Thanks, this worked!